CMakeLists.txt

```diff
@@ -50,6 +50,13 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
 endif(NOT CMAKE_BUILD_TYPE)
 
+# Parse repository info from git if available
+include(version)
+
+configure_file(
+  ${CMAKE_SOURCE_DIR}/src/version.hpp.in
+  ${CMAKE_BINARY_DIR}/src/version.hpp
+  )
```

Comment:
For release tarballs, this will need special treatment, e.g. shipping version.hpp inside the tarball and skipping this configure_file.

Reply:
I think an install statement for the generated header should do it.

Reply:
Also at the moment it will just work, the version.hpp is not needed at runtime.